pkg/ignition/builder.go

```diff
@@ -103,6 +105,10 @@ func (b *ignitionBuilder) Generate() ([]byte, error) {
 		"/etc/NetworkManager/dispatcher.d/01-hostname",
 		0744, false,
 		[]byte("[[ \"$DHCP6_FQDN_FQDN\" =~ \".\" ]] && hostnamectl set-hostname --static --transient $DHCP6_FQDN_FQDN")))
+	config.Storage.Files = append(config.Storage.Files, ignitionFileEmbed(
+		"/etc/NetworkManager/dispatcher.d/02-hostname",
+		0744, false,
+		[]byte("hostnamectl set-hostname --static --transient b.hostname")))
```

Comment:
I don't think we can set --static, that seems to take precedence over DHCP (it updates `/etc/hostname`).

From what I can gather we want to do `set-hostname --transient` whenever the current hostname [is localhost](https://github.com/systemd/systemd/blob/main/src/basic/hostname-util.c#L183-L192).

I suggest we consolidate this into the 01-hostname script,

Reply:
Updated the 01-hostname script. Not sure if this is what we want the order of the 2 operations to be.